presto-main/src/main/java/com/facebook/presto/operator/scalar/SessionFunctions.java

```diff
@@ -41,9 +40,6 @@ public static Slice currentUser(ConnectorSession session)
     @SqlType(StandardTypes.VARCHAR)
     public static Slice currentPath(ConnectorSession session)
     {
-        if (session.getPath() == null) {
-            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Session path is null");
-        }
-        return utf8Slice(session.getPath());
+        return utf8Slice(((FullConnectorSession) session).getSession().getPath().toString());
```

Comment:
Add a comment that `current_path` is an language construct, and has special access to Presto internals.